### PR TITLE
Removed redundant cast

### PR DESCRIPTION
**Pull request by matbech**
The cast is redundant.

Trivial change.

### CODE REVIEW

File: src/Http/Segment.php [hunk 1]
```diff
@@ -446,7 +446,7 @@ public function getAssembledParams()
      */
     protected function encode(string $value)
     {
-        $key = (string) $value;
+        $key = $value;
```
Comment (Xerkus) [inline src/Http/Segment.php]: @matbech separate `$key` and `$value` variables are redundant as well now.

**Review by Xerkus — CHANGES_REQUESTED** [review]
This is a patch level change, it should be done against 3.4.x branch for the next patch level release. Please rebase.

Also, for your commit to be included we require commits to be signed off for [Developer Certificate of Origin](https://developercertificate.org). Please see DCO check.

**Comment by matbech** [comment]
Please go ahead with the required changes. The whole process (requirements) of doing a trivial change is ridiculous.

**Comment by matbech** [comment]
Thank you

**Pull request merged**